Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/GestureBubblingTests.cs

```diff
@@ -0,0 +1,199 @@
+
+				RunningApp.TapCoordinates(rect.CenterX, rect.CenterY);
+			}
+			else
+			{
+				RunningApp.Tap(q => q.Marked(menuItem));
+			}
+
+			// Find the start label
+			RunningApp.WaitForElement(q => q.Marked("Start"));
+
+			// Find the control we're testing
+			var result = RunningApp.WaitForElement(q => q.Marked(TargetAutomationId));
+			var target = result.First().Rect;
+
+			// Tap the control
+			var y = target.CenterY;
+			var x = target.CenterX;
+
+			// In theory we want to tap the center of the control. But Stepper lays out differently than the other controls,
+			// so we need to adjust for it until someone fixes it
+			if (menuItem == "Stepper")
+			{
+				y = target.Y + 5;
+				x = target.X + 5;
+			}
+
+			RunningApp.TapCoordinates(x, y);
+
+			if (menuItem == nameof(DatePicker) || menuItem == nameof(TimePicker))
+			{
+				// These controls show a pop-up which we have to cancel/done out of before we can continue
+#if __ANDROID__
+				var cancelButtonText = "Cancel";
+#elif __IOS__
+				var cancelButtonText = "Done";
+#else
+				var cancelButtonText = "X";
+#endif
+				RunningApp.WaitForElement(q => q.Marked(cancelButtonText));
+				RunningApp.Tap(q => q.Marked(cancelButtonText));
+			}
+
+			if (frameShouldRegisterTap)
+			{
+				RunningApp.WaitForElement(q => q.Marked("Frame was tapped"));
+			}
+			else
+			{
+				RunningApp.WaitForElement(q => q.Marked("Start"));
+			}
+		}
+#endif
+
+		ContentPage CreateTestPage(View view)
+		{
+			var instructions = new Label();
+
+			if (_controlsWhichShouldAllowTheTapToBubbleUp.Contains(view.GetType().Name))
+			{
+				instructions.Text =
+					"Tap the frame below. The label with the text 'No taps yet' should change its text to 'Frame was tapped'.";
+			}
+			else
+			{
+				instructions.Text =
+					"Tap the frame below. The label with the text 'No taps yet' should not change.";
+			}
+
+			var label = new Label { Text = "Start" };
+
+			var frame = new Frame { Content = new StackLayout { Children = { view } } };
+
+			var rec = new TapGestureRecognizer { NumberOfTapsRequired = 1 };
+			rec.Tapped += (s, e) => { label.Text = "Frame was tapped"; };
+			frame.GestureRecognizers.Add(rec);
+			
+			var layout = new StackLayout();
+		
+			layout.Children.Add(instructions);
+			layout.Children.Add(label);
+			layout.Children.Add(frame);
+
+			return new ContentPage { Content = layout };
+		}
+
+		Button MenuButton(string label, Func<View> view)
+		{
+			var button = new Button { Text = label };
+
+			var testView = view();
+			testView.AutomationId = TargetAutomationId;
+
+			button.Clicked += (sender, args) => PushAsync(CreateTestPage(testView));
+
+			return button;
+		}
+
+		// These controls should allow the tap gesture to bubble up to their container; everything else should absorb the gesture
+		readonly List<string> _controlsWhichShouldAllowTheTapToBubbleUp = new List<string> { nameof(Image), nameof(Label), nameof(BoxView), nameof(Frame) };
```

Comment:
I reformatted it to be a more friendly width. AFAIK there's no way to get GH to use the full screen width unless you use a custom stylesheet or something.